redis/src/main/java/org/apache/calcite/adapter/redis/RedisDataBase.java

```diff
@@ -0,0 +1,145 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.redis;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Set;
+
+import redis.clients.jedis.Jedis;
+
+/**
+ * The abstract class with RedisDataBase.
+ */
+public abstract class RedisDataBase {
+  String tableName;
+  String dataFormat;
+  String keyDelimiter;
+  RedisDataType dataType = null;
+  RedisDataFormat redisDataFormat = null;
+  List<LinkedHashMap<String, Object>> fields;
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  void init(Jedis jedis, RedisTableFieldInfo tableFieldInfo) {
+    String type = jedis.type(tableFieldInfo.getTableName());
+    fields = tableFieldInfo.getFields();
+    dataFormat = tableFieldInfo.getDataFormat();
+    tableName = tableFieldInfo.getTableName();
+    keyDelimiter = tableFieldInfo.getKeyDelimiter();
+    dataType = RedisDataType.fromTypeName(type);
+    redisDataFormat = RedisDataFormat.fromTypeName(tableFieldInfo.getDataFormat());
+    objectMapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true)
+        .configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true)
+        .configure(JsonParser.Feature.ALLOW_COMMENTS, true);
+    assert redisDataFormat != null;
+    assert dataType != null;
+  }
+
+  public List<Object[]> read(Jedis jedis, RedisTableFieldInfo tableFieldInfo) {
+    return read(jedis, tableFieldInfo);
+  }
+
+  private Object[] parseJson(String value) {
+    assert StringUtils.isNotEmpty(value);
+    Object[] arr = new Object[fields.size()];
+    try {
+      JsonNode jsonNode = objectMapper.readTree(value);
+      Object obj;
+      for (int i = 0; i < arr.length; i++) {
+        obj = fields.get(i).get("mapping");
+        if (obj == null) {
+          arr[i] = "";
+        } else {
+          arr[i] = jsonNode.findValue(fields.get(i).get("mapping").toString());
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Parsing json failed: ", e);
+    }
+    return arr;
+  }
+
+  private Object[] parseCsv(String value) {
+    assert StringUtils.isNotEmpty(value);
+    String[] values = value.split(keyDelimiter);
+    Object[] arr = new Object[fields.size()];
+    assert values.length == arr.length;
+    try {
+      for (int i = 0; i < arr.length; i++) {
+        arr[i] = values[i] == null ? "" : values[i];
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    return arr;
+  }
+
+  List<Object[]> parse(Jedis jedis, Set<String> keys) {
```

Comment:
`Jedis` instance is available in `init()` method of current object. Can it be re-used ?